buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ValidateRestSpecPlugin.java

```diff
@@ -39,6 +39,16 @@ public void apply(Project project) {
                 }));
                 task.setJsonSchema(new File(project.getRootDir(), "rest-api-spec/src/main/resources/schema.json"));
             });
-        project.getTasks().named("precommit").configure(t -> t.dependsOn(validateRestSpecTask));
+
+        Provider<ValidateJsonNoKeywordsTask> validateNoKeywordsTask = project.getTasks()
+            .register("validateNoKeywords", ValidateJsonNoKeywordsTask.class, task -> {
+                task.setInputFiles(Util.getJavaTestAndMainSourceResources(project, filter -> {
```

Comment:
I wonder if it makes sense to add an `mustRunAfter` ordering rule so that we run this after validating the schema. Essentially, if the spec format is busted we are likely to have issues running this task.